### ISSUE TITLE
Upload failed - no error log generated

### ISSUE THREAD

**Issue by mostlyjason** [LycheeOrg/Lychee-v3#180]
### Detailed description of the problem
The site runs normally and I can login and view photos. However, when I try to upload a photo I occasionally get a error saying the upload failed. I see a 500 return code in my browser developer tools, but there doesn't appear to be an error message. It seems to give this error after reaching 100% but before the image is processed. I then restart the docker container, which restarts the server, and I can upload photos successfully for a period of time.

The log output doesn't seem to have any relevant error messages. If a server returns a 500 error shouldn't we see that in the logs? 

### Steps to reproduce the issue
See above. I'm not aware of the exact steps because it happens after some time has passed.

### Output of the diagnostics (Settings => Diagnostics)
I tried uploading the Lychee-front package but it doesn't seem to have any effect on the problem. I doubt uploads/small is a problem because it happens inconsistently.

Diagnostics
-----------
Error: 'uploads/small' is missing or has insufficient read/write privileges

System Information
------------------
/usr/share/webapps/lychee/Lychee-front/package.json not found. Please do: git submodule init ; git submodule update
DB Version:             update_030210
System:                 Linux
PHP Version:            7.2
MySQL Version:          50505
Imagick:                1
Imagick Active:         1
Imagick Version:        1799
GD Version:             bundled (2.1.0 compatible)
Plugins:                

### Browser and system
Firefox
linuxserver/docker-lychee

**Comment by ildyria** [comment]
Very weird.

I just created a wiki page: https://github.com/LycheeOrg/Lychee/wiki/I-have-an-Error-500-what-can-I-do%3F
If you can do that, hopefully you will get something better than a white screen of death at your next crash. :/

**Comment by mostlyjason** [comment]
Ok just turned on the error log will try to reproduce. It doesnt give a white screen of death just a failure message in the upload modal window.

**Comment by ildyria** [comment]
Press F12, and go do your network panel in your browser. You should see a call to `php/index.php` you can have a look at the *response*, this will give you more info of why it crashes.

**Comment by mostlyjason** [comment]
Ok here are the error messages. They don't look very helpful unfortunately. It's too bad there isn't any logging we can turn on that might give more information.

This is from the index.php response:
```
<html>
<head><title>500 Internal Server Error</title></head>
<body bgcolor="white">
<center><h1>500 Internal Server Error</h1></center>
<hr><center>nginx/1.14.2</center>
</body>
</html>
```
This is from the console output:
```
Object { description: "Upload failed. Server returned an unkown error!", params: XMLHttpRequest, response: "" }
main.js:2955:2
lychee.error
http://mysite:8080/dist/main.js:2955:2
process/xhr.onload
http://mysite:8080/dist/main.js:5503:27
```

**Comment by ildyria** [comment]
It is rather surprising. Can you upload the picture that generate this error?
Error 500 are the hardest to debug because they mean that php crashed. :s
Do you have access to php or apache2 logs files (probably in /var/log/ ... ) ?

**Comment by mostlyjason** [comment]
Ok I figured it out. It was tricky because the linuxserver image redirected the error logs to /config/log/nginx/error.log instead of the usual /var/log or standard out.  I saw an error message saying

```
2019/02/03 18:47:54 [crit] 287#287: *107 open() "/var/tmp/nginx/client_body/0000000001" failed (13: Permission denied), client: 172.24.0.10, server: _, request: "POST /php/index.php HTTP/1.1", host: "mysite", referrer: "https://mysite/php/index.php"
```

The alpine nignx image doesn't make the /var/tmp/nginx directory writable for some reason. The fix is to change to container initialization script to make it writable. I'll send a separate PR to the linuxserver repo.
